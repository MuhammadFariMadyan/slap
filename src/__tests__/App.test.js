import React from "react";
import { shallow } from "enzyme";
import configureMockStore from "redux-mock-store";

import Connected, { App } from "../App";

jest.mock("../components/container/Home", () => "Home");
jest.mock("../components/container/Editor", () => "Editor");

describe("App", () => {
  const mockStore = configureMockStore();

  const setup = propOverrides => {
    const props = {
      ...propOverrides
    };

    const wrapper = shallow(<App {...props} />);

    return {
      wrapper,
      props
    };
  };

  const setupConnected = storeOverrides => {
    const store = mockStore({
      activeProjectID: null,
      ...storeOverrides
    });
    const wrapper = shallow(<Connected store={store} />);

    return {
      wrapper,
      store
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should renders as connected", () => {
    const { wrapper } = setupConnected();

    expect(wrapper).toMatchSnapshot();
  });

  it("should renders with active project", () => {
    const { wrapper } = setup({
      activeProjectID: "project-1"
    });

    expect(wrapper).toMatchSnapshot();
  });

  it("should map new to props", () => {
    const { wrapper } = setupConnected();

    expect(wrapper.props()).toMatchObject({
      activeProjectID: null
    });
  });
});
