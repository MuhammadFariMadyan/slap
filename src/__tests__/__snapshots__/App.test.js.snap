// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should renders as connected 1`] = `
<Component
  activeProjectID={null}
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`App should renders default 1`] = `
<styled.div>
  <Home />
</styled.div>
`;

exports[`App should renders with active project 1`] = `
<styled.div>
  <Editor />
</styled.div>
`;
